lib/sles4sap_publiccloud.pm

```diff
@@ -78,9 +89,145 @@ sub get_promoted_hostname {
+    record_info("cmd_out", $cmd_out);
+    my @all_parameters = map { if (/^Hosts/) { s,Hosts/,,; s,",,g; $_ } else { () } } split("\n", $cmd_out);
+    my @all_hosts = uniq map { (split("/", $_))[0] } @all_parameters;
+
+    for my $host (@all_hosts) {
+        my %host_parameters = map { my ($node, $parameter, $value) = split(/[\/=]/, $_);
+            if ($host eq $node) { ($parameter, $value) } else { () } } @all_parameters;
+        push(@topology, \%host_parameters);
+
+        if (defined($hostname) && $hostname eq $host) {
+            return \%host_parameters;
+        }
+    }
+
+    return \@topology;
+}
+
+=head2 is_hana_online
+    is_hana_online([timeout => 120, wait_for_start => 'false']);
+
+    Check if hana DB is online. Define 'wait_for_start' to wait for DB to start.
+
+=cut
+
+sub is_hana_online {
+    my ($self, %args) = @_;
+    my $wait_for_start = defined($args{wait_for_start}) ? 1 : 0;
+    my $timeout = bmwqemu::scale_timeout($args{timeout} // 120);
+    my $start_time = time;
+    my $consecutive_passes = 0;
+    my $db_status;
+
+    while ($consecutive_passes < 3) {
+        $db_status = $self->get_replication_info()->{online} eq "true" ? 1 : 0;
+        return $db_status if $wait_for_start == 0;
+
+        # Reset pass counter in case of fail.
+        $consecutive_passes = $db_status ? ++$consecutive_passes : 0;
+        die("DB did not start within defined timeout: $timeout s") if (time - $start_time > $timeout);
+        sleep 10;
+    }
+    return $db_status;
+}
+
+=head2 is_hana_resource_running
+    is_hana_resource_running([timeout => 60]);
+
+    Checks if resource msl_SAPHana_* is running on given node.
+=cut
+
+sub is_hana_resource_running {
+    my ($self) = @_;
+    my $hostname = $self->{my_instance}->{instance_id};
+    my $hana_resource = join("_",
+        "msl",
+        "SAPHana",
+        "HDB",
+        get_required_var("INSTANCE_SID") . get_required_var("INSTANCE_ID"));
+
+    my $resource_output = $self->run_cmd(cmd => "crm resource status " . $hana_resource, quiet => 1);
+    my $node_status = grep /is running on: $hostname/, $resource_output;
+    record_info("Node status", "$hostname: $node_status");
+    return $node_status;
+}
+
+=head2 stop_hana
+    stop_hana([timeout => $timeout, method => $method]);
+
+    Stops HANA database using default or specified method.
+    "stop" - stops database using "HDB stop" command.
+    "kill" - kills database processes using "HDB -kill" command.
+    "crash" - crashes entire os using "/proc-sysrq-trigger" method.
+=cut
+
+sub stop_hana {
+    my ($self, %args) = @_;
+    my $timeout = bmwqemu::scale_timeout($args{timeout} // 300);
+    my $method = defined($args{method}) ? $args{method} : 'stop';
+    my %commands = (
+        stop => "HDB stop",
+        kill => "HDB kill -x",
+        crash => "echo b > /proc/sysrq-trigger &"
+    );
+
+    my $cmd = $commands{$method};
```

Comment:
Good idea in case somebody uses sub outside of this test. 
Will add:
```
croak("HANA stop method '$args{method}' unknown.") unless exists $args{$method};
```

Reply:
Ah yes... correct